source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -506,16 +506,22 @@ void CacheFilter::handleCacheHit(bool end_stream_after_headers) {
 void CacheFilter::handleCacheHitWithRangeRequest() {
   if (!lookup_result_->range_details_.has_value()) {
     ENVOY_LOG(error, "handleCacheHitWithRangeRequest() should not be called without "
-                     "range_details being populated in lookup_result_");
+                     "range_details_ being populated in lookup_result_");
     return;
   }
+  ENVOY_BUG(lookup_result_->content_length_.has_value(),
```

Comment:
If the cache has received enough of the response to serve the requested range, it seems fine to do so. It just means we'll have to serve `content-range: begin-end/*`. (If you just don't want to implement and test it, that's fine, but please then leave it as a todo for either of us.)

Reply:
Done.
A bit of weirdness there with respect to trailers - I made a somewhat arbitrary decision that 206 responses should not include trailers, based on:
1. I couldn't find any spec commentary either way on the subject.
2. range requests are usually for file downloads which wouldn't have trailers anyway.
3. even if they did have trailers you probably wouldn't want to receive the trailers repeatedly for each range request.
4. pragmatically we don't even have the information as to whether there will be trailers or not when filling a range request prematurely.